src/data/classification_dataloader.py

```diff
@@ -0,0 +1,191 @@
+
+    config = self._params
+    tokenizer = tf_text.BertTokenizer(config.vocab_filename,
+                                      lower_case=True,
+                                      preserve_unused_token=True,
+                                      token_out_type=tf.int32)
+
+    use_rand_aug = config.use_rand_aug
+    is_training = config.is_training
+    input_patterns = config.input_path.split(',')
+    data_utils.check_input_patterns(input_patterns)
+
+    batch_size_per_replica = input_context.get_per_replica_batch_size(
+        config.global_batch_size)
+
+    dataset = tf.data.Dataset.list_files(input_patterns,
+                                         shuffle=is_training,
+
+                                         seed=config.seed)
+    if is_training:
+      dataset = dataset.repeat()
+
+      # We set shuffle buffer to exactly match total number of
+      # training files to ensure that training data is well shuffled.
+      input_files = []
+      for input_pattern in input_patterns:
+        input_files.extend(tf.io.gfile.glob(input_pattern))
+      dataset = dataset.shuffle(len(input_files))
+
+    if (input_context and input_context.num_input_pipelines > 1):
+      dataset = dataset.shard(input_context.num_input_pipelines,
+                              input_context.input_pipeline_id)
+
+    dataset = dataset.interleave(
+        tf.data.TFRecordDataset,
+        cycle_length=config.cycle_length,
+        num_parallel_calls=tf.data.experimental.AUTOTUNE)
+
+    # Sets shuffle buffer size larger than # records in a single shard.
+    if is_training:
+      dataset = dataset.shuffle(4096)
+
+    # Gets mapping functions.
+    decode_fn = data_utils.get_decode_fn(self.name_to_features,
+                                         tokenizer,
+                                         config,
+                                         merge_dims=True,
+                                         is_training=is_training,
+                                         keep_unnormalized_patch_embeddings=False,
+                                         use_rand_aug=use_rand_aug)
+
+    # Makes sure batch size for matching function is also larger than
+    # negative_positive_ratio to avoid creating false negatives examples.
+    max_shift = config.negative_positive_ratio + config.min_shift
+    # Use enough batch size for creating negative examples.
+    batch_size_in_matching_fn = ((max_shift // batch_size_per_replica + 2) *
+                                 batch_size_per_replica)
+    matching_fn = data_utils.get_matching_fn(config,
+                                             batch_size_in_matching_fn,
+                                             config.negative_positive_ratio)
+
+    word_ids_fn = data_utils.get_word_ids_fn(config)
+    split_features_fn = data_utils.get_split_features_fn(input_keys, label_keys)
+
+    # TODO(roylu): relative_pos_max_distance and relative_att_num_core_layers 
```

Comment:
I just reviewed another CL with this exact same code and comment. Is this copy/pasted code? id there is shared logic, it might be better to capture it into a helper function